src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -80,6 +80,33 @@ public interface ProgrammingExerciseRepository extends JpaRepository<Programming
     @EntityGraph(type = LOAD, attributePaths = "submissionPolicy")
     Optional<ProgrammingExercise> findWithSubmissionPolicyById(Long exerciseId);
 
+    List<ProgrammingExercise> findByProjectKey(String projectKey);
```

Comment:
shouldn't the methods named findAll since they return a list?

Reply:
I renamed it to be clearer. There's no difference between `findBy` and `findAllBy` functionality wise.